packages/dds/tree/src/feature-libraries/optional-field/optionalField.ts

```diff
@@ -718,69 +520,58 @@ export const optionalChangeHandler: FieldChangeHandler<
-
-	// The node that is moved into the field (if any).
-	const nodeMovedIntoField = change.valueReplace?.src;
-
-	return change.childChanges.map(([register, nodeId]) => {
-		// The node is removed in the input context iif register is not self.
-		const inputIndex = register === "self" ? 0 : undefined;
-		const outputIndex =
-			register === "self"
-				? // If the node starts out as not-removed, it is removed in the output context iff the field content is changed
-					isFieldContentChanged
-					? undefined
-					: 0
-				: // If the node starts out as removed, then it remains removed in the output context iff it is not the node that is moved into the field
-					!areEqualRegisterIdsOpt(register, nodeMovedIntoField)
-					? undefined
-					: 0;
-		return [nodeId, inputIndex, outputIndex];
-	});
-}
+function getCrossFieldKeys(change: OptionalChangeset): CrossFieldKeyRange[] {
+	const keys: CrossFieldKeyRange[] = [];
+	if (change.valueReplace !== undefined) {
+		keys.push({
+			key: { ...change.valueReplace.dst, target: CrossFieldTarget.Source },
+			count: 1,
+		});
 
-function* relevantRemovedRoots(
-	change: OptionalChangeset,
-	relevantRemovedRootsFromChild: RelevantRemovedRootsFromChild,
-): Iterable<DeltaDetachedNodeId> {
-	const alreadyYielded = new RegisterMap<boolean>();
-
-	for (const [src] of change.moves) {
-		if (!alreadyYielded.has(src)) {
-			alreadyYielded.set(src, true);
-			yield nodeIdFromChangeAtom(src);
+		if (change.valueReplace.src !== undefined && change.valueReplace.src !== "self") {
+			keys.push({
+				key: { ...change.valueReplace.src, target: CrossFieldTarget.Destination },
+				count: 1,
+			});
 		}
 	}
 
-	for (const [id, childChange] of change.childChanges) {
-		// Child changes make the tree they apply to relevant unless that tree existed in the starting context of
-		// of this change.
-		if (id !== "self" && !alreadyYielded.has(id)) {
-			alreadyYielded.set(id, true);
-			yield nodeIdFromChangeAtom(id);
-		}
-		yield* relevantRemovedRootsFromChild(childChange);
+	return keys;
+}
+
+function getNestedChanges(change: OptionalChangeset): NestedChangesIndices {
+	if (change.childChange === undefined) {
+		return [];
 	}
 
-	const selfSrc = change.valueReplace?.src;
-	if (selfSrc !== undefined && selfSrc !== "self" && !alreadyYielded.has(selfSrc)) {
-		yield nodeIdFromChangeAtom(selfSrc);
+	return [[change.childChange, 0]];
+}
+
+/**
+ * Helper function for use in rebase which returns child change for the base changeset.
+ */
+function getBaseChildChange(
+	baseChange: OptionalChangeset,
+	nodeManager: RebaseNodeManager,
+): NodeId | undefined {
+	const attachId = getEffectfulDst(baseChange.valueReplace);
+	if (attachId !== undefined) {
+		const movedChange = nodeManager.getNewChangesForBaseAttach(attachId, 1).value?.nodeChange;
+		if (movedChange !== undefined) {
+			assert(baseChange.childChange === undefined, "Unexpected child change");
+			return movedChange;
+		}
 	}
```

Comment:
I'm having hard time understanding what's going on here. Things that would help:
- Docs on RebaseNodeManager methods
- Better docs on this function
- Doc within the function
- Perhaps using different terminology might help (e.g., "inbound" and "outbound")
- If there is a bug (seems that way, but hard to tell), then fixing it would make the rest easier to understand